test/e2e_node/device_plugin.go

```diff
@@ -68,27 +68,39 @@ func testDevicePlugin(f *framework.Framework, enablePluginWatcher bool, pluginSo
 			initialConfig.FeatureGates[string(features.KubeletPodResources)] = true
 		})
 		It("Verifies the Kubelet device plugin functionality.", func() {
-			By("Start stub device plugin")
-			// fake devices for e2e test
-			devs := []*pluginapi.Device{
-				{ID: "Dev-1", Health: pluginapi.Healthy},
-				{ID: "Dev-2", Health: pluginapi.Healthy},
+			By("Wait for node is ready to start with")
+			Eventually(func() bool {
+				node, err := f.ClientSet.CoreV1().Nodes().Get(framework.TestContext.NodeName, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+				for _, cond := range node.Status.Conditions {
+					if cond.Type == v1.NodeReady && cond.Status == v1.ConditionTrue {
+						return true
+					}
+				}
+				return false
+			}, 5*time.Minute, framework.Poll).Should(BeTrue())
+			dp := framework.SampleDevicePlugin()
+			for i := range dp.Spec.Containers[0].Env {
+				if dp.Spec.Containers[0].Env[i].Name == envVarNamePluginSockDir {
+					dp.Spec.Containers[0].Env[i].Value = pluginSockDir
+				}
 			}
-
-			socketPath := pluginSockDir + "dp." + fmt.Sprintf("%d", time.Now().Unix())
-			framework.Logf("socketPath %v", socketPath)
-
-			dp1 := dm.NewDevicePluginStub(devs, socketPath, resourceName, false)
-			dp1.SetAllocFunc(stubAllocFunc)
```

Comment:
maybe also delete `stubAllocFunc` from this file?

Reply:
done